eventlet/websocket.py

```diff
@@ -135,7 +135,12 @@ def __call__(self, environ, start_response):
         return wsgi.ALREADY_HANDLED
 
     def _handle_legacy_request(self, environ):
-        sock = environ['eventlet.input'].get_socket()
+        if 'eventlet.input' in environ.keys():
+            sock = environ['eventlet.input'].get_socket()
+        elif 'gunicorn.socket' in environ.keys():
+            sock = environ["gunicorn.socket"]
+        else:
+            raise("No knows sockets availabe.")
```

Comment:
- Use one kind of quotes (single).
- `raise` needs exception type. I don't have a better idea than `raise Exception('helpful text')`.
- Provide helpful message in exception. Something along `environ must contain either 'eventlet.input' object with .get_socket() method or 'guniron.socket'`.